### ISSUE TITLE
Isnt AMDGPU missing OpenGL ?

### ISSUE THREAD

**Issue by luan-cestari** [Tids/mhwd-addon-amdgfx-hwe#2]
Hi,

I saw the code from this repository and also from the https://github.com/manjaro/mhwd . I thought if it would be better if the installer set the OpenGL to the so that  AMD built as it is using Mesa with Gallium now. Or It could have the option to change to the AMD built and the default is the Mesa. 

Also, The script is calling setgl and it is no longer a valid parameter https://github.com/manjaro/mhwd/commit/370f531c8ee9833e5d45bb8fec7d0744b15b2292#diff-dd01fea3b74f983ae774c2eeeeca78ba 

Thanks in advance

**Comment by Tids** [comment]
You're right with this. This repo isn't up-to-date anymore.
The newer code is in
https://github.com/Tids/packages-community/tree/mhwd-addon-amdgfx-hwe

I don't understand the OpenGL part. Mesa is *the* OpenGL provider for all amd generations. the only exception is the fglrx driver. But this one has a complete own infrastructure and is not compatible with the normal Manjaro setup.

**Comment by luan-cestari** [comment]
Hi @Tids 

Do think it is better if I open the same issue on the other repo (like yours or even the https://github.com/manjaro/packages-community )? I saw that you and philmmanjaro are maintaining the repo, good work =) .

About the OpenGL part,  maybe I'm wrong but when I was using Ubuntu with amdgpu-pro I think the OpenGL vendor was made by AMD, it wasn't Gallium with Mesa. 

If in there are extra drivers like OpenGL and so on in amdgpu-pro (not the open-source amdgpu, the one that only supports rhel, Ubuntu and SLED http://support.amd.com/en-us/kb-articles/Pages/AMDGPU-PRO-Driver-for-Linux-Release-Notes.aspx ) , is it possible to include in this project? 

One thing I can say about performance which makes me think that on Ubuntu with amdgpu-pro had a different OpenGL implementation is the performance on a game I was testing, with Gallium right now I got sometimes half of FPS (sometimes the difference is even bigger) . The only differences that comes to my mind between my current OS (manjaro) and Ubuntu are the amdgpu-pro (which got the installer from the amd website) and the amd-microcode package (https://packages.debian.org/search?keywords=amd64-microcode) (which as far as I understood it maybe comes with linux-firmware on manjaro).

Below some output about the OpenGL implementation that I'm using right now:
```
$ glxinfo | grep "OpenGL"
OpenGL vendor string: X.Org
OpenGL renderer string: Gallium 0.4 on AMD HAWAII (DRM 3.8.0 / 4.9.24-1-MANJARO, LLVM 4.0.0)
OpenGL core profile version string: 4.5 (Core Profile) Mesa 17.0.4
OpenGL core profile shading language version string: 4.50
OpenGL core profile context flags: (none)
OpenGL core profile profile mask: core profile
OpenGL core profile extensions:
OpenGL version string: 3.0 Mesa 17.0.4
OpenGL shading language version string: 1.30
OpenGL context flags: (none)
OpenGL extensions:
OpenGL ES profile version string: OpenGL ES 3.1 Mesa 17.0.4
OpenGL ES profile shading language version string: OpenGL ES GLSL ES 3.10
OpenGL ES profile extensions:
```
```
$ glxgears -info | grep GL_VERSION
GL_VERSION    = 3.0 Mesa 17.0.4
```

Thank you

**Comment by Tids** [comment]
> with amdgpu-pro I think the OpenGL vendor was made by AMD, it wasn't Gallium with Mesa.

Yep. AMDGPU-Pro has an own implementation to bring OpenGL (and ES) support in. It does not use Mesa in any way.

> is it possible to include in this project?

No. AMDGPU-Pro is propritary, We dont have the sources and we can not include it because of the license. But even if we could, AMDGPU-Pro depends on the Ubuntu Kernels and does not work better (in terms of OpenGL support) than radeonsi (the Mesa OpenGL implementation). For Example Feral Interactive only supports RadeonSI, no the AMDGPU-Pro OpenGL.

>  the amd-microcode package (https://packages.debian.org/search?keywords=amd64-microcode) (which as far as I understood it maybe comes with linux-firmware on manjaro).

Thats true

>`$ glxgears -info | grep GL_VERSION`
> `GL_VERSION    = 3.0 Mesa 17.0.4`

Dont grep for the OpenGL version, but the Core Profile. Mesa dont supports Legacy functionalities of OpenGL Compat. With OpenGL3.x there was a switch. Old functionalities where sorted out and the to something that is OpenGL Compat and new functions where going in OpenGL core profiles. Mesa does not wanted to support a compat version, so thes only support the Core Profiles.

Thats not a bug or does something bad. All OpenGL applications/Engines/Framework are using the core Profile.

**Comment by luan-cestari** [comment]
Hi,

You are right, I made new set of tests after discovering that the radeon-profile and the cpu governor ( cpupower frequency-set -g) was holding back the performance. I can say it double the performance in comparison with Ubuntu with amdgpu-pro, using wine galium enabled. I also figure out to change my refresh rate of the screen with xrandr.

Thanks a lot.
